File: backend/engines/engine_metadata.py
"""V67 — Internal Metadata Watermark (IMW)
Central author/watermark metadata for backend engine.
"""

AUTHOR_NAME = "Onazi Treasure"
WATERMARK_SHORT = "OJ"
ENGINE_SIGNATURE = "PATHFINDER_ENGINE_CORE_OJ"
VERSION = "V68"

def get_metadata() -> dict:
    return {
        "author": AUTHOR_NAME,
        "watermark": WATERMARK_SHORT,
        "engine_signature": ENGINE_SIGNATURE,
        "version": VERSION,
    }

File: frontend/src/engines/EngineMetadata.ts
// V67 — Internal Metadata Watermark (IMW)
import { AUTHOR_NAME, WATERMARK_SHORT, ENGINE_SIGNATURE, getMetadata } from '../services/watermark';

export interface EngineMetadata {
  author: string;
  watermark: string;
  engine_signature: string;
  version: string;
}

export const ENGINE_METADATA: EngineMetadata = getMetadata();

// Expose in diagnostics only during development (not rendered in UI)
if (import.meta.env.DEV) {
  (window as any).PATHMAP_METADATA = ENGINE_METADATA;
}
